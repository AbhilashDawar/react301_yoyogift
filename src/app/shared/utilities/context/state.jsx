import * as React from 'react';

let ContextState = React.createContext();

let initialState = {
    showLoader: false,
    // filterParams: {
    //     startValue: config.rangeDefaults.min,
    //     endValue: config.rangeDefaults.max,
    //     discountOption: 0
    // }
};

let reducer = (state, action) => {
    switch (action.type) {
        case "RESET":
            return initialState;
        case "SHOW":
            return { ...state, showLoader: true };
        case "HIDE":
            return { ...state, showLoader: false };
        // case "UPDATE_FILTERS":
        //     return { ...state, filterParams: action.payload };
        default:
            return { ...state };
    }
};

function ContextStateProvider(props) {
    let [state, dispatch] = React.useReducer(reducer, initialState);
    let value = { state, dispatch };

    return (
        <ContextState.Provider value={value}>{props.children}</ContextState.Provider>
    );
}

let ContextStateConsumer = ContextState.Consumer;

export { ContextState, ContextStateProvider, ContextStateConsumer };