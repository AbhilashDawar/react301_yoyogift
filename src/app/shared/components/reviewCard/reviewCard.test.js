import React from 'react';
import ReviewCard from './index.jsx';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { ContextStateProvider } from '../../utilities/context/state.jsx';

test('ReviewCard Component', () => {
    let props = {
        review: {
            userName: 'name',
            rating: 2,
            review: 'review',
        }
    };
    const tree = renderer.create(
        <MemoryRouter>
            <ContextStateProvider>
                <ReviewCard review={props.review} />
            </ContextStateProvider>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});