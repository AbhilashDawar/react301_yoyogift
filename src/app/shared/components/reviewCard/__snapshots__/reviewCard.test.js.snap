// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-card-1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root makeStyles-sections-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4c-4.41 0-8 3.59-8 8 0 1.82.62 3.49 1.64 4.83 1.43-1.74 4.9-2.33 6.36-2.33s4.93.59 6.36 2.33C19.38 15.49 20 13.82 20 12c0-4.41-3.59-8-8-8zm0 9c-1.94 0-3.5-1.56-3.5-3.5S10.06 6 12 6s3.5 1.56 3.5 3.5S13.94 13 12 13z"
              opacity=".3"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root makeStyles-userName-4 MuiTypography-body1 MuiTypography-gutterBottom"
          >
            name
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
      >
        <div
          className="MuiGrid-root makeStyles-sections-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <span
              aria-label="2 Stars"
              className="MuiRating-root MuiRating-readOnly"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              role="img"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              className="MuiTypography-root makeStyles-reviewText-3 MuiTypography-body1 MuiTypography-gutterBottom"
            >
              review
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
